sycl/unittests/Extensions/CommandGraph.cpp

```diff
@@ -361,3 +361,189 @@ TEST_F(CommandGraphTest, RecordSubGraph) {
       sycl::detail::getSyclObjImpl(Node3MainGraph));
   ASSERT_EQ(Queue.get_context(), MainGraphExecImpl->getContext());
 }
+
+TEST_F(CommandGraphTest, InOrderQueue) {
+  sycl::property_list Properties{sycl::property::queue::in_order()};
+  sycl::queue InOrderQueue{Dev, Properties};
+
+  // Record in-order queue with three nodes
+  Graph.begin_recording(InOrderQueue);
+  auto Node1Graph = InOrderQueue.submit(
+      [&](sycl::handler &cgh) { cgh.single_task<class TestKernel>([]() {}); });
+
+  auto PtrNode1 = sycl::detail::getSyclObjImpl(Graph)->getLastInorderNode(
+      sycl::detail::getSyclObjImpl(InOrderQueue));
+  ASSERT_NE(PtrNode1, nullptr);
+  ASSERT_TRUE(PtrNode1->MPredecessors.empty());
+
+  auto Node2Graph = InOrderQueue.submit(
+      [&](sycl::handler &cgh) { cgh.single_task<class TestKernel>([]() {}); });
+
+  auto PtrNode2 = sycl::detail::getSyclObjImpl(Graph)->getLastInorderNode(
+      sycl::detail::getSyclObjImpl(InOrderQueue));
+  ASSERT_NE(PtrNode2, nullptr);
+  ASSERT_NE(PtrNode2, PtrNode1);
+  ASSERT_EQ(PtrNode1->MSuccessors.size(), 1lu);
+  ASSERT_EQ(PtrNode1->MSuccessors.front(), PtrNode2);
+  ASSERT_EQ(PtrNode2->MPredecessors.size(), 1lu);
+  ASSERT_EQ(std::shared_ptr<experimental::detail::node_impl>(
+                PtrNode2->MPredecessors.front()),
+            PtrNode1);
+
+  auto Node3Graph = InOrderQueue.submit(
+      [&](sycl::handler &cgh) { cgh.single_task<class TestKernel>([]() {}); });
+
+  auto PtrNode3 = sycl::detail::getSyclObjImpl(Graph)->getLastInorderNode(
+      sycl::detail::getSyclObjImpl(InOrderQueue));
+  ASSERT_NE(PtrNode3, nullptr);
+  ASSERT_NE(PtrNode3, PtrNode2);
+  ASSERT_EQ(PtrNode2->MSuccessors.size(), 1lu);
+  ASSERT_EQ(PtrNode2->MSuccessors.front(), PtrNode3);
+  ASSERT_EQ(PtrNode3->MPredecessors.size(), 1lu);
+  ASSERT_EQ(std::shared_ptr<experimental::detail::node_impl>(
+                PtrNode3->MPredecessors.front()),
+            PtrNode2);
+
+  Graph.end_recording(InOrderQueue);
+
+  auto GraphExec = Graph.finalize();
```

Comment:
Does it make sense to also check the schedule created by `finalize()` similar to other unitests? This would verify the execution order of the executable graph as well.

Reply:
Thanks for this feedback. I added a check on the created schedule for each new test.